src/common/snippets/src/pass/validate.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/pass/validate.hpp"
+
+#include "snippets/op/convert_saturation.hpp"
+#include "snippets/op/convert_truncation.hpp"
+#include "snippets/itt.hpp"
+
+#include "openvino/op/fake_quantize.hpp"
+#include "openvino/op/broadcast.hpp"
+#include "openvino/op/matmul.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/transpose.hpp"
+#include "openvino/op/softmax.hpp"
+#include "openvino/core/validation_util.hpp"
+
+
+namespace ov {
+namespace snippets {
+namespace pass {
+
+namespace {
+#define VALIDATE(op, op_type, validator)    \
+    if (ov::is_type<op_type>(op)) \
+        OPENVINO_ASSERT(validator(op), "Snippets validation of OV body has been failed: " + \
+                                       std::string(op->get_type_name()) + " op " + op->get_friendly_name() + " is not supported");
```

Comment:
You probably need `else` here as well (as was in my comment). Otherwise there will be a bunch of redundant casts for every successfully validated op

Reply:
For sure, I didn't notice `else` in your comment. And unfortunately, I didn't add it myself. Thanks!